### PR TITLE
Validate storage when invoked. Allow other middlewares to start session.

### PR DESCRIPTION
**Pull request by ivandokov**
Fixes https://github.com/slimphp/Slim-Csrf/issues/53

### CODE REVIEW

File: src/Guard.php [hunk 1]
```diff
@@ -88,19 +88,7 @@ public function __construct(
...
-        } else {
-            if (!isset($_SESSION)) {
-                throw new RuntimeException('CSRF middleware failed. Session not found.');
-            }
-            if (!array_key_exists($prefix, $_SESSION)) {
-                $_SESSION[$prefix] = [];
-            }
-            $this->storage = &$_SESSION[$prefix];
-        }
+        $this->storage = &$storage;
```
Comment (akrabat) [inline src/Guard.php]: I'm not 100% sure that we want to do this if `$storage` is an instance of `ArrayAccess` as it's already passed by reference.

File: src/Guard.php [hunk 2]
```diff
@@ -181,22 +194,22 @@ public function generateToken()
 
         return $this->keyPair;
     }
-
```
Comment (akrabat) [inline src/Guard.php]: Please revert all these whitespace changes.

**Comment by akrabat** [comment]
@ivandokov: Are you likely to address the issues I noted?

**Comment by ivandokov** [comment]
I totally forgot about this. I will make an update later today or tomorrow.

File: src/Guard.php (hunk 1 above)
Comment (ivandokov) [inline src/Guard.php]: If the check is only for the `$_SESSION` in `validateStorage` we will have two places to set `$this->storage`. I think it is better to keep it in a single method. Also passing by reference twice is not an issue in the case of `ArrayAccess`.

**Pull request merged**